specs/Workers.md

```diff
@@ -0,0 +1,1402 @@
+                  }
+
+                  return S_OK;
+              })
+              .Get(),
+          &m_dedicatedWorkerCreatedToken));
+    }
+```
+
+## Service Worker
+## Monitoring the Registration/Destruction of Service Workers
+
+The following example demonstrates how to subscribe to the event that is raised
+when a web page registers a service worker. This event provides a
+`CoreWebView2ServiceWorkerRegistration` object, enabling the host application to
+interact with the service workers via upcoming PostMessage and
+WorkerMessageReceived APIs.
+
+## .NET/WinRT
+```c#
+    void PostMessageToWorker(CoreWebView2ServiceWorker worker)
+    {
+        // The service worker communicates updates back to the host application,
+        // such as when resource caching is complete. The host application can
+        // listen for the WorkerMessageReceived event to receive these updates
+        // from the service worker.
+        worker.WorkerMessageReceived += (sender, args) =>
+        {
+            var message = args.TryGetWorkerMessageAsString;
+
+            // Process the messages received from the worker. Depending on the
+            // content of these messages, you might choose to log the data for
+            // debugging, update the user interface to reflect changes, or raise
+            // other actions within your application.
+        };
+
+        // You can send messages to the service worker using its PostMessage method.
+        // The host application is sending a message to the worker to cache certain
+        // resources. The message is a JSON string that specifies the type of the
+        // message and the payload of the message.
+        worker.PostMessage("{ MyMessage: { \"type\": \"CACHE_URLS\", \"payload\": [\"/styles/main.css\", \"/scripts/main.js\", \"/images/logo.png\"]} }");
+    }
+
+    CoreWebView2ServiceWorkerManager ServiceWorkerManager_;
+    void ServiceWorkerRegisteredExecuted(object target, ExecutedRoutedEventArgs e)
+    {
+        if (ServiceWorkerManager_ != nullptr)
+        {
+            ServiceWorkerManager_ = WebViewProfile.ServiceWorkerManager;
+        }
+        ServiceWorkerManager_.ServiceWorkerRegistered += async (sender, args) =>
+        {
+            CoreWebView2ServiceWorkerRegistration serviceWorkerRegistration = args.ServiceWorkerRegistration;
+            if(serviceWorkerRegistration != null)
+            {
+                serviceWorkerRegistration.WorkerDestroyed += (sender, args) =>
+                {
+                    /*Cleanup on worker destruction*/
+                };
+
+                LogMessage("Service worker has been registered with a scope: " + serviceWorkerRegistration.ScopeUri);
+
+                serviceWorkerRegistration.ServiceWorkerActivated += async (sender, args) =>
+                {
+                    CoreWebView2ServiceWorker serviceWorker = args.ActiveServiceWorker;
+                    serviceWorker.WorkerDestroyed += (sender, args) =>
```

Comment:
Two events for worker destroyed, and listening to them both?

Reply:
See above.